enter.c

```diff
@@ -36,68 +37,59 @@
 #define lengthof(Arr) (sizeof (Arr) / sizeof (*Arr))
 
 struct cloneflag {
-	const char *name;
 	int flag;
+	const char *proc_ns_name;
 };
 
-enum {
-
-	/* To ensure backward and forward compatibility for the ability of
-	   unsharing the maximum of namespaces, we re-define these constants. */
-	BST_CLONE_NEWNET    = 0x40000000,
-	BST_CLONE_NEWUTS    = 0x04000000,
-	BST_CLONE_NEWCGROUP = 0x02000000,
-	BST_CLONE_NEWNS     = 0x00020000,
-	BST_CLONE_NEWPID    = 0x20000000,
-	BST_CLONE_NEWUSER   = 0x10000000,
-	BST_CLONE_NEWIPC    = 0x08000000,
-	BST_CLONE_NEWTIME   = 0x00000080,
-
-	ALL_NAMESPACES = 0
-		| BST_CLONE_NEWCGROUP
-		| BST_CLONE_NEWIPC
-		| BST_CLONE_NEWNS
-		| BST_CLONE_NEWNET
-		| BST_CLONE_NEWPID
-		| BST_CLONE_NEWUSER
-		| BST_CLONE_NEWUTS
-		| BST_CLONE_NEWTIME
-		,
+static struct cloneflag flags[] = {
+	{ BST_CLONE_NEWCGROUP, "cgroup" },
+	{ BST_CLONE_NEWIPC,    "ipc",   },
+	{ BST_CLONE_NEWNS,     "mnt"    },
+	{ BST_CLONE_NEWNET,    "net"    },
+	{ BST_CLONE_NEWPID,    "pid"    },
+	{ BST_CLONE_NEWTIME,   "time"   },
+	{ BST_CLONE_NEWUSER,   "user"   },
+	{ BST_CLONE_NEWUTS,    "uts"    },
 };
 
-static int opts_to_unshareflags(const struct entry_settings *opts)
+const char *nsname(int ns)
 {
-	static struct cloneflag flags[] = {
-		{ "cgroup",  BST_CLONE_NEWCGROUP },
-		{ "ipc",     BST_CLONE_NEWIPC },
-		{ "mount",   BST_CLONE_NEWNS },
-		{ "network", BST_CLONE_NEWNET },
-		{ "pid",     BST_CLONE_NEWPID },
-		{ "time",    BST_CLONE_NEWTIME },
-		{ "user",    BST_CLONE_NEWUSER },
-		{ "uts",     BST_CLONE_NEWUTS },
-		{ "all",     ALL_NAMESPACES },
-	};
+	return flags[ns].proc_ns_name;
+}
+	
+
+struct nsaction {
+	/* Which namespace is being unshared or entered */
+	int flag;
+
+	/* If enter_fd is -1, then the namespace given by "flag" is to
```

Comment:
No longer accurate, maybe say that it can be >=0 or either NSACTION_*?

Reply:
yep